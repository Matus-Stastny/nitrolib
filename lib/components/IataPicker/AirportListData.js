"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactRelay = require("react-relay");

var _Alert = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Alert"));

var _environment = _interopRequireDefault(require("../../services/environment"));

var _AirportList = _interopRequireDefault(require("./AirportList"));

var _Translate = _interopRequireDefault(require("../Translate"));

var _graphql;

var _ref2 =
/*#__PURE__*/
React.createElement(_Alert.default, {
  type: "critical"
}, React.createElement(_Translate.default, {
  t: "common.api_error"
}));

var AirportListData = function AirportListData(_ref) {
  var value = _ref.value,
      onSelect = _ref.onSelect,
      environment = _ref.environment;
  return React.createElement(_reactRelay.QueryRenderer, {
    environment: environment,
    query: _graphql || (_graphql = function _graphql() {
      var node = require("./__generated__/AirportListDataQuery.graphql");

      if (node.hash && node.hash !== "ac5d0d52123d1c5371ab83d2f14a7146") {
        console.error("The definition of 'AirportListDataQuery' appears to have changed. Run `relay-compiler` to update the generated files to receive the expected data.");
      }

      return require("./__generated__/AirportListDataQuery.graphql");
    }),
    variables: {
      input: value
    },
    render: function render(res) {
      if (res.error) {
        return _ref2;
      }

      if (!res.props || !res.props.allLocations) {
        return null;
      } // $FlowExpected: TODO describe


      return React.createElement(_AirportList.default, {
        list: res.props.allLocations,
        onSelect: onSelect
      });
    }
  });
};

AirportListData.defaultProps = {
  environment: _environment.default
};
var _default = AirportListData;
exports.default = _default;