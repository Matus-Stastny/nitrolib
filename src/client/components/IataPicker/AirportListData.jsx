// @flow strict
import * as React from "react";
import { graphql, QueryRenderer } from "react-relay";

import environment from "client/services/environment";
import AirportList from "./AirportList";

type Props = {|
  value: string,
  onSelect: (id: string) => void,
  environment: typeof environment,
|};

const AirportListData = (props: Props) => (
  <QueryRenderer
    environment={props.environment}
    query={graphql`
      query AirportListDataQuery($input: String!) {
        allLocations(search: $input, options: { locationType: airport }) {
          ...AirportList_list
        }
      }
    `}
    variables={{ input: props.value }}
    render={res => {
      if (res.error) {
        // TODO <Alert />
        return <span>Error :(</span>;
      }

      if (!res.props) {
        return null;
      }

      return <AirportList list={res.props.allLocations} onSelect={props.onSelect} />;
    }}
  />
);

AirportListData.defaultProps = {
  environment,
};

export default AirportListData;
