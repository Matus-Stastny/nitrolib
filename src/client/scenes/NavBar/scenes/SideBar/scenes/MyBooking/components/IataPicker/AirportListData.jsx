// @flow strict
import * as React from "react";
import { graphql, QueryRenderer } from "react-relay";

import environment from "client/services/environment";
import AirportList from "../AirportList/index";

type Props = {|
  value: string,
  onSelect: (id: string) => void,
|};

const AirportListData = (props: Props) => (
  <QueryRenderer
    environment={environment}
    query={graphql`
      query AirportListDataQuery($input: String!) {
        allLocations(search: $input, options: { locationType: airport }) {
          ...AirportList_list
        }
      }
    `}
    variables={{ input: props.value }}
    render={res => {
      if (res.error) {
        // TODO <Alert />
        return <span>Error :(</span>;
      }

      if (!res.props) {
        return null;
      }

      return <AirportList list={res.props.allLocations} onSelect={props.onSelect} />;
    }}
  />
);

export default AirportListData;
