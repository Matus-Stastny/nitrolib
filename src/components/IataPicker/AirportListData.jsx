// @flow strict
import * as React from "react";
import { graphql, QueryRenderer } from "react-relay";
import Alert from "@kiwicom/orbit-components/lib/Alert";

import environmentReal from "../../services/environment";
import AirportList from "./AirportList";
import Translate from "../Translate";

type Props = {|
  value: string,
  onSelect: (id: string) => void,
  environment: typeof environmentReal,
|};

const AirportListData = ({ value, onSelect, environment }: Props) => (
  <QueryRenderer
    environment={environment}
    query={graphql`
      query AirportListDataQuery($input: String!) {
        allLocations(search: $input, options: { locationType: airport }) {
          ...AirportList_list
        }
      }
    `}
    variables={{ input: value }}
    render={res => {
      if (res.error) {
        return (
          <Alert type="critical">
            <Translate t={__("common.api_error")} />
          </Alert>
        );
      }

      if (!res.props || !res.props.allLocations) {
        return null;
      }

      // $FlowExpected: TODO describe
      return <AirportList list={res.props.allLocations} onSelect={onSelect} />;
    }}
  />
);

AirportListData.defaultProps = {
  environment: environmentReal,
};

export default AirportListData;
