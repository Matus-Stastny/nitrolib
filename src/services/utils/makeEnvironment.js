// @flow strict
import { Environment, Network, RecordSource, Store } from "relay-runtime";

export type Input = {
  query: string,
  variables: { [key: string]: string },
};

export type Call = (input: Input) => Promise<any>;

const makeFetchQuery = (call: Call) => (operation, variables) =>
  call({
    query: operation.text,
    variables,
  });

const store = new Store(new RecordSource());

const makeEnvironment = (call: Call) =>
  new Environment({
    network: Network.create(makeFetchQuery(call)),
    store,
  });

export default makeEnvironment;
