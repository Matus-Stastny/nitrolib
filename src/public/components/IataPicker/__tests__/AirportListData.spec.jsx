// @flow strict
import * as React from "react";
import { mount } from "enzyme";

import makeEnvironment from "public/services/utils/makeEnvironment";
import AirportListData from "../AirportListData";

const res = {
  data: {
    allLocations: {
      edges: [
        {
          node: {
            locationId: "VIE",
            name: "Vienna airport",
            type: "airport",
            city: {
              name: "Vienna",
            },
            country: {
              locationId: "at",
            },
          },
        },
      ],
    },
  },
};

describe("#AirportListData", () => {
  test("render error", async () => {
    const promise = Promise.reject(new Error("error"));
    const environment = makeEnvironment(() => promise);

    const wrapper = mount(
      <AirportListData value="VIE" onSelect={jest.fn()} environment={environment} />,
    );

    await promise.catch(() => null);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("render loading", () => {
    const promise = Promise.resolve(res);
    const environment = makeEnvironment(() => promise);

    const wrapper = mount(
      <AirportListData value="VIE" onSelect={jest.fn()} environment={environment} />,
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("render results", async () => {
    const promise = Promise.resolve(res);
    const environment = makeEnvironment(() => promise);

    const wrapper = mount(
      <AirportListData value="VIE" onSelect={jest.fn()} environment={environment} />,
    );

    await promise;

    expect(wrapper.html()).toMatchSnapshot();
  });
});
