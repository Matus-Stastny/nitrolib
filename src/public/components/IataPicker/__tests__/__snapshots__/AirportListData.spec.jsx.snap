// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#AirportListData render error 1`] = `"<span>Error :(</span>"`;

exports[`#AirportListData render loading 1`] = `null`;

exports[`#AirportListData render results 1`] = `"<button class=\\"AirportResult__Container-s1fcv9cu-0 szXVj\\"><img class=\\"Flag__Img-s15sfv5w-0 hGAKSQ\\" src=\\"//images.kiwi.com/flags/32x32/at.png\\" alt=\\"at\\">Vienna (VIE)</button>"`;
